pkg/util/iptables/iptables.go

```diff
@@ -529,12 +491,79 @@ func (runner *runner) checkRuleUsingCheck(args []string) (bool, error) {
 	return false, fmt.Errorf("error checking rule: %v: %s", err, out)
 }
 
+const (
+	// Max time we wait for an iptables flush to complete after we notice it has started
+	iptablesFlushTimeout = 5 * time.Second
+	// How often we poll while waiting for an iptables flush to complete
+	iptablesFlushPollTime = 100 * time.Millisecond
+)
+
+// Monitor is part of Interface
+func (runner *runner) Monitor(canary Chain, tables []Table, reloadFunc func(), interval time.Duration, stopCh <-chan struct{}) {
+	for {
+		for _, table := range tables {
+			if _, err := runner.EnsureChain(table, canary); err != nil {
+				klog.Warningf("Could not set up iptables canary %s/%s: %v", string(table), string(canary), err)
+				return
+			}
+		}
+
+		// Poll until stopCh is closed or iptables is flushed
+		err := utilwait.PollUntil(interval, func() (bool, error) {
+			if runner.chainExists(tables[0], canary) {
+				return false, nil
+			}
+			klog.V(2).Infof("iptables canary %s/%s deleted", string(tables[0]), string(canary))
+
+			// Wait for the other canaries to be deleted too before returning
+			// so we don't start reloading too soon.
+			err := utilwait.PollImmediate(iptablesFlushPollTime, iptablesFlushTimeout, func() (bool, error) {
+				for i := 1; i < len(tables); i++ {
+					if runner.chainExists(tables[i], canary) {
+						return false, nil
+					}
+				}
+				return true, nil
+			})
+			if err != nil {
+				klog.Warning("Inconsistent iptables state detected.")
+			}
+			return true, nil
+		}, stopCh)
+
+		for _, table := range tables {
```

Comment:
Do we need to delete it here?  Best case they are already gone as per the loop above.  Worst case they are not gone but we timed out and are proceeding anyway.  We'll just recreate them at the top of this loop.

This seems like a waste